components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -440,6 +941,9 @@ macro_rules! cast_as_duration {
                     let result = Duration::parse($extra, extra.ret_field_type.get_decimal() as i8);
                     match result {
                         Ok(dur) => Ok(Some(dur)),
+                        // TODO, here is not same as TiDB's because
+                        //  1. Duration::parse is not same as TiDB's impl
+                        //  2. if parse return err, tidb can get res, too, however, we can't.
                         Err(e) => match e.code() {
```

Comment:
There is no problem to handle it like this and the final result will be consistent.